packages/vite/src/node/optimizer/index.ts

```diff
@@ -1216,8 +1216,8 @@ export function getDepHash(config: ResolvedConfig, ssr: boolean): string {
       assetsInclude: config.assetsInclude,
       plugins: config.plugins.map((p) => p.name),
       optimizeDeps: {
-        include: optimizeDeps?.include,
-        exclude: optimizeDeps?.exclude,
+        include: Array.from(new Set(optimizeDeps?.include)).sort(),
+        exclude: Array.from(new Set(optimizeDeps?.exclude)).sort(),
```

Comment:
Nice one! Sounds good to me. We should still keep it as undefined if the option wasnt set though to avoid changing the hash for all the projects without explicit include/exclude and trigger a cold start for them